packages/core/src/network/heartbeat.ts

```diff
@@ -159,10 +165,15 @@ export default class Heartbeat {
 
     finished = true
 
-    for (const pingResult of pingResults) {
-      // Filter unexpected network errors
+    for (let [resultIndex, pingResult] of pingResults.entries()) {
+      await setImmediate()
       if (pingResult instanceof Error) {
-        continue
+        // we need to get the destination so we can map a ping error properly
+        const [destination, _abortSignal] = pingWork[resultIndex]
+        pingResult = {
```

Comment:
This can have side effects. It maybe better to use `if/else` and call `networkPeers.updateRecord` twice.

Reply:
What side-effect are you talking about?

Reply:
Good point. Adapted.